### PR TITLE
Feature/custom pipeline

### PR DESCRIPTION
**Pull request by CrypticCabub**
Beginning to add in a custom pipeline for automated testing of the solution

Pipeline is currently deployable via `cdk deploy`, pulls code from github, and runs the junit/pytest unit tests. Deployment test currently fails and is being worked on

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/infrastructure/bin/aws-instance-scheduler.ts [hunk 1]
```diff
@@ -56,4 +57,5 @@ new AwsInstanceSchedulerRemoteStack(app, 'aws-instance-scheduler-remote', {
     solutionVersion: SOLUTION_VERSION,
     appregApplicationName: APP_REG_APPLICATION_TYPE,
     appregSolutionName: APP_REG_SOLUTION_NAME
-});
\ No newline at end of file
+});
+new PipelineStack(app, 'testing-pipeline');
```
Comment (hearde) [inline source/infrastructure/bin/aws-instance-scheduler.ts:61]: Isn't this the name of the stack in cfn? Shouldn't it be a bit more descriptve? e.g. `InstanceSchedulerTestingPipeline`?

**Review by hearde — APPROVED** [review]

File: source/infrastructure/pipeline/pipeline-stack.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      installCommands: [
+        'pip install tox',
+        'tox -e cdk',
+        'tox -e lambda -- --junitxml=deployment/test-reports/lambda-test-report.xml'
+      ],
+      commands: [
+        'cd source/infrastructure',
+        "npx cdk synth",
+        'cd ../../deployment',
+        'ls -r',
```
[the remark this answers is not in the document]
Comment (CrypticCabub) [reply source/infrastructure/pipeline/pipeline-stack.ts:90]: that was some testing code that I forgot to delete, will remove it. thanks

File: source/infrastructure/bin/aws-instance-scheduler.ts (hunk 1 above)
Comment (CrypticCabub) [reply source/infrastructure/bin/aws-instance-scheduler.ts:61]: it appears to be applied as a prefix before a CDK Pipeline hash that keeps it unique. a more specific name is probably a good idea.  I had originally been trying to keep it generic for reusability, but I don't think this pipeline code is quite there yet

File: source/infrastructure/pipeline/pipeline-stack.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+  constructor(scope: Construct, construct_id: string) {
+    super(scope, construct_id);
+  }
+
+  instanceSchedulerStack = new AwsInstanceSchedulerStack(this, STACK_NAME, {
+    appregApplicationName: "",
+    appregSolutionName: "",
+    description: "",
+    solutionBucket: "",
+    solutionId: "",
```
[the remark this answers is not in the document]
Comment (CrypticCabub) [reply source/infrastructure/pipeline/pipeline-stack.ts:146]: not really. in my local develop I'm working on that part of the pipeline next and copied over some of the values. However, we may need to be careful with appreg and metrics reporting -- don't want our testing pipelines to report hundreds of fake deploys

**Review by tmekari — APPROVED** [review]

**Pull request merged**